File: IntersectionOfTwoLinkedLists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        ListNode aPointer = headA;
        ListNode bPointer = headB;
        
        while(aPointer != bPointer){
            if(aPointer == null){
                aPointer = headB;
            }else{
                aPointer = aPointer.next;
            }
            if(bPointer == null){
                bPointer = headA;
            }else{
                bPointer = bPointer.next;
            }
        }
        return aPointer;
    }
}

File: IntersectionOfTwoLinkedLists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} headA
 * @param {ListNode} headB
 * @return {ListNode}
 */
var getIntersectionNode = function(headA, headB) {
    let headAPointer = headA;
    let headBPointer = headB;
    while(headAPointer !== headBPointer) {
        if(headAPointer) headAPointer = headAPointer.next;
        else headAPointer = headB;
        if(headBPointer) headBPointer = headBPointer.next;
        else headBPointer = headA;
    }
    return headAPointer;
};
